Fix NuValidator image

alecslupu opened the pull request:
#### :tophat: What? Why?
It seems that NuValidator image is broken, or introduced new rules that we have not fixed yet. This fixes the validator to the latest working version : sha:7667b0ffa6d395c27aa8f9e21db1cfe6b66549245a3972e9397d255de5ef0ec6


```

  1) Explore versions when showing version behaves like accessible page passes HTML validation
     Failure/Error: DEFAULT_FAILURE_NOTIFIER = lambda { |failure, _opts| raise failure }

       ######
       line 103: The “aria-labelledby” attribute must not be specified on any “div” element unless the element has a “role” value other than “caption”, “code”, “deletion”, “emphasis”, “generic”, “insertion”, “paragraph”, “presentation”, “strong”, “subscript”, or “superscript”.

           </label>

       >>    <div id="accordion-trigger-essential" data-controls="accordion-panel-essential" aria-labelledby="accordion-title-essential">
             <h3 id="accordion-title-essential" class="cookies__category-trigger-title">
               Essential
             </h3>

       ######
       line 198: The “aria-labelledby” attribute must not be specified on any “div” element unless the element has a “role” value other than “caption”, “code”, “deletion”, “emphasis”, “generic”, “insertion”, “paragraph”, “presentation”, “strong”, “subscript”, or “superscript”.

           </label>

       >>    <div id="accordion-trigger-preferences" data-controls="accordion-panel-preferences" aria-labelledby="accordion-title-preferences">
             <h3 id="accordion-title-preferences" class="cookies__category-trigger-title">
               Preferences
             </h3>

       ######
```

#### Testing
1. The pipeline should be green

:hearts: Thank you!

alecslupu committed:
Fix NuValidator image

andreslucena commented:
@alecslupu please DO provide a link with a failure run, a stacktrace, etc - so it is easy for me to actually review this.

These are the things that I'd have expected: 

Link to a failure run:  https://github.com/decidim/decidim/actions/runs/20109790292/job/57703683652?pr=15682

## Stacktrace (as an example, there are more)

```
  1) Explore versions when showing version behaves like accessible page passes HTML validation
     Failure/Error: DEFAULT_FAILURE_NOTIFIER = lambda { |failure, _opts| raise failure }

       ######
       line 103: The “aria-labelledby” attribute must not be specified on any “div” element unless the element has a “role” value other than “caption”, “code”, “deletion”, “emphasis”, “generic”, “insertion”, “paragraph”, “presentation”, “strong”, “subscript”, or “superscript”.

           </label>

       >>    <div id="accordion-trigger-essential" data-controls="accordion-panel-essential" aria-labelledby="accordion-title-essential">
             <h3 id="accordion-title-essential" class="cookies__category-trigger-title">
               Essential
             </h3>

       ######
       line 198: The “aria-labelledby” attribute must not be specified on any “div” element unless the element has a “role” value other than “caption”, “code”, “deletion”, “emphasis”, “generic”, “insertion”, “paragraph”, “presentation”, “strong”, “subscript”, or “superscript”.

           </label>

       >>    <div id="accordion-trigger-preferences" data-controls="accordion-panel-preferences" aria-labelledby="accordion-title-preferences">
             <h3 id="accordion-title-preferences" class="cookies__category-trigger-title">
               Preferences
             </h3>

       ######
       line 229: The “aria-labelledby” attribute must not be specified on any “div” element unless the element has a “role” value other than “caption”, “code”, “deletion”, “emphasis”, “generic”, “insertion”, “paragraph”, “presentation”, “strong”, “subscript”, or “superscript”.

           </label>

       >>    <div id="accordion-trigger-analytics" data-controls="accordion-panel-analytics" aria-labelledby="accordion-title-analytics">
             <h3 id="accordion-title-analytics" class="cookies__category-trigger-title">
               Analytics and statistics
             </h3>

       ######
       line 260: The “aria-labelledby” attribute must not be specified on any “div” element unless the element has a “role” value other than “caption”, “code”, “deletion”, “emphasis”, “generic”, “insertion”, “paragraph”, “presentation”, “strong”, “subscript”, or “superscript”.

           </label>

       >>    <div id="accordion-trigger-marketing" data-controls="accordion-panel-marketing" aria-labelledby="accordion-title-marketing">
             <h3 id="accordion-title-marketing" class="cookies__category-trigger-title">
               Marketing
             </h3>

       ######
       line 366: An “li” element that is a descendant of a “role=menu” element or “role=menubar” element must not have any “role” value other than “group”, “menuitem”, “menuitemcheckbox”, “menuitemradio”, or “separator”.

           <div id="dropdown-menu-language-chooser" aria-hidden="true">
             <ul class="main-footer__language" role="menu">
       >>          <li class="text-black text-md hover:bg-secondary hover:text-white is-active" role="presentation">
                   <a role="menuitem" lang="en" class="p-2 w-full block" rel="nofollow" data-method="post" href="/locale?locale=en">English</a>
                 </li>
                 <li class="text-black text-md hover:bg-secondary hover:text-white " role="presentation">

       ######
       line 369: An “li” element that is a descendant of a “role=menu” element or “role=menubar” element must not have any “role” value other than “group”, “menuitem”, “menuitemcheckbox”, “menuitemradio”, or “separator”.

                   <a role="menuitem" lang="en" class="p-2 w-full block" rel="nofollow" data-method="post" href="/locale?locale=en">English</a>
                 </li>
       >>          <li class="text-black text-md hover:bg-secondary hover:text-white " role="presentation">
                   <a role="menuitem" lang="ca" class="p-2 w-full block" rel="nofollow" data-method="post" href="/locale?locale=ca">Català</a>
                 </li>
                 <li class="text-black text-md hover:bg-secondary hover:text-white " role="presentation">

       ######
       line 372: An “li” element that is a descendant of a “role=menu” element or “role=menubar” element must not have any “role” value other than “group”, “menuitem”, “menuitemcheckbox”, “menuitemradio”, or “separator”.

                   <a role="menuitem" lang="ca" class="p-2 w-full block" rel="nofollow" data-method="post" href="/locale?locale=ca">Català</a>
                 </li>
       >>          <li class="text-black text-md hover:bg-secondary hover:text-white " role="presentation">
                   <a role="menuitem" lang="es" class="p-2 w-full block" rel="nofollow" data-method="post" href="/locale?locale=es">Castellano</a>
                 </li>
             </ul>

       ######
       line 438: An “li” element that is a descendant of a “role=menu” element or “role=menubar” element must not have any “role” value other than “group”, “menuitem”, “menuitemcheckbox”, “menuitemradio”, or “separator”.

             <div id="dropdown-menu-language-chooser-mobile" aria-hidden="true">
               <ul class="main-footer__language" role="menu">
       >>            <li class="text-black text-md hover:bg-secondary hover:text-white is-active" role="presentation">
                     <a role="menuitem" lang="en" class="p-2 w-full block" rel="nofollow" data-method="post" href="/locale?locale=en">English</a>
                   </li>
                   <li class="text-black text-md hover:bg-secondary hover:text-white " role="presentation">

       ######
       line 441: An “li” element that is a descendant of a “role=menu” element or “role=menubar” element must not have any “role” value other than “group”, “menuitem”, “menuitemcheckbox”, “menuitemradio”, or “separator”.

                     <a role="menuitem" lang="en" class="p-2 w-full block" rel="nofollow" data-method="post" href="/locale?locale=en">English</a>
                   </li>
       >>            <li class="text-black text-md hover:bg-secondary hover:text-white " role="presentation">
                     <a role="menuitem" lang="ca" class="p-2 w-full block" rel="nofollow" data-method="post" href="/locale?locale=ca">Català</a>
                   </li>
                   <li class="text-black text-md hover:bg-secondary hover:text-white " role="presentation">

       ######
       line 444: An “li” element that is a descendant of a “role=menu” element or “role=menubar” element must not have any “role” value other than “group”, “menuitem”, “menuitemcheckbox”, “menuitemradio”, or “separator”.

                     <a role="menuitem" lang="ca" class="p-2 w-full block" rel="nofollow" data-method="post" href="/locale?locale=ca">Català</a>
                   </li>
       >>            <li class="text-black text-md hover:bg-secondary hover:text-white " role="presentation">
                     <a role="menuitem" lang="es" class="p-2 w-full block" rel="nofollow" data-method="post" href="/locale?locale=es">Castellano</a>
                   </li>
               </ul>

     [Screenshot Image]: file:///home/runner/work/decidim/decidim/spec/decidim_dummy_app/tmp/screenshots/failures_r_spec_example_groups_explore_versions_when_showing_version_behaves_like_accessible_page_passes_html_validation_321.png

     [Screenshot HTML]: file:///home/runner/work/decidim/decidim/spec/decidim_dummy_app/tmp/screenshots/failures_r_spec_example_groups_explore_versions_when_showing_version_behaves_like_accessible_page_passes_html_validation_321.html


     Shared Example Group: "accessible page" called from ./spec/system/debates_versions_spec.rb:57
     # /home/runner/work/decidim/decidim/decidim-dev/lib/decidim/dev/test/rspec_support/accessibility_examples.rb:132:in 'block (2 levels) in <top (required)>'

Finished in 11 minutes 0 seconds (files took 15.8 seconds to load)
362 examples, 1 failure

Failed examples:

rspec ./spec/system/debates_versions_spec.rb[1:3:1:2] # Explore versions when showing version behaves like accessible page passes HTML validation
```

So, the rules have changed - do you plan to fix it in another PR? Should we at least create an issue to keep track of this? If not I'm afraid is the typical thing that there will be more rules in the future with newer versions and the technical debt will keep growing.  

On the other hand, I see that you've implemented this fix at #15682 - I guess you want to have it granular in one PR so it is easier to backport, right? 

---

FYI - all of this metadata, questions, stacktrace, links, etc that I'm providing is what I'd expect to see in this template:

<img width="540" height="335" alt="image" src="https://github.com/user-attachments/assets/299c5345-9528-49f6-8beb-d6fa6b113e81" />

Changing a line and adding a phrase but without actually giving a thought on what implications this have to maintainability/releases/pipelines is like doing 50% of the actual work.

andreslucena reviewed (commented):
Not blocking this so we can move forward with other PRs, but please check out my last comment: https://github.com/decidim/decidim/pull/15773#issuecomment-3640867891

andreslucena commented:
After the merge I realized, that we should have titled this as "Pin NuValidator image to old version" to better reflect that this isn't a "fix", just a quick hack to keep going (but that it is actually adding technical debt , as it is a temporal workaround to the actual issue that is "the validation rules have changed").

alecslupu commented:
> On the other hand, I see that you've implemented this fix at #15682 - I guess you want to have it granular in one PR so it is easier to backport, right?

I have implemented the fix in #15682 to make sure the pipeline does not fail due to changes we do in the API. 
Also, i have implemented as a small PR to make sure that we can backport if we need to.

alecslupu commented:
> So, the rules have changed - do you plan to fix it in another PR? Should we at least create an issue to keep track of this? If not I'm afraid is the typical thing that there will be more rules in the future with newer versions and the technical debt will keep growing.

I feel that the changed rules are related with what @Stef-Rousset is doing in #14951 , #14886 , #14944, so yeah ... i will hve a lookj when i will manage to pick up those

alecslupu commented:
Adding `no-backport` as we do not have to backport it in previous versions.
